src/frontend/js/components/SaleTunnel/components/SaleTunnelStepResume/index.tsx

```diff
@@ -18,34 +21,66 @@ const messages = defineMessages({
     description: "Text to remind that order's invoice will be send by email soon",
     id: 'components.SaleTunnelStepResume.successDetailMessage',
   },
+  successDetailSignatureMessage: {
+    defaultMessage: 'In order to enroll to course runs you first need to sign the contract.',
+    description: 'Text to remind that order needs to be signed',
+    id: 'components.SaleTunnelStepResume.successDetailSignatureMessage',
+  },
   cta: {
     defaultMessage: 'Start this course now!',
     description: 'Label to the call to action to close sale tunnel',
     id: 'components.SaleTunnelStepResume.cta',
   },
+  ctaSignature: {
+    defaultMessage: 'Go to signature',
+    description: 'Label to the call to action to close sale tunnel if there is a pending signature',
+    id: 'components.SaleTunnelStepResume.ctaSignature',
+  },
 });
 
 interface SaleTunnelStepResumeProps {
   next: () => void;
 }
 
-export const SaleTunnelStepResume = ({ next }: SaleTunnelStepResumeProps) => (
-  <section className="SaleTunnelStepResume">
-    <header className="SaleTunnelStepResume__header">
-      <SuccessIcon />
-      <h3 className="SaleTunnelStepResume__title">
-        <FormattedMessage {...messages.congratulations} />
-      </h3>
-    </header>
-    <p className="SaleTunnelStepResume__content">
-      <FormattedMessage {...messages.successMessage} />
-      <br />
-      <FormattedMessage {...messages.successDetailMessage} />
-    </p>
-    <footer className="SaleTunnelStepResume__footer">
-      <Button onClick={next}>
-        <FormattedMessage {...messages.cta} />
-      </Button>
-    </footer>
-  </section>
-);
+export const SaleTunnelStepResume = ({ next }: SaleTunnelStepResumeProps) => {
```

Comment:
WDYT about spliting into two components: `SaleTunnelStepResume` / `SaleTunnelStepResumeWithContract`

Reply:
I thought the opposite, not so much duplication but easier to maintain both step as they are quite different.

Reply:
Can you provide an example? As there are only two `if (...)` conditions I think having two components is a bit heavy, but maybe you have short approach in mind